draft-specifications/Notation.md

```diff
@@ -0,0 +1,214 @@
+Concrete Representation of messages.
+This document describes both the notation and representation.
+The [CapTP Specification](CapTP%20Specification.md) and [Model](Model.md) documents
+employ the abstract notation to imply the conrete representation.
+
+The concrete representation is a binary encoding that is occasionally and
+incidentally human-readable.
+
+The abstract notation is human-readable and human-writable and is not intended
+for machine processing.
+The abstract notation does not attempt to comprehensively cover all expressible
+concrete messages, but is suitable for expressing all the specification and
+examples.
+
+> The abstract notation comes from
+> [Preserves](https://preserves.gitlab.io/preserves/TUTORIAL.html).
+> The concrete representation is a subset of
+> [Syrup](https://github.com/ocapn/syrup).
+
+## Notation of the Notation
+
+For the grammars of both the abstract notation and concrete representations,
+this document uses the following notation:
+
+- `literal`
+- adjacent adjacent
+- alternative / alternative
+- zero or more *
+- one or more +
+- first - last
+- optional ?
+- ( group )
+- _reference_
+- _referent_ : _definition_
+- :: semantics
+
+Both grammars allow for the interpolation of arbitrary [ASCII]
+(https://datatracker.ietf.org/doc/html/rfc20) space (SP), tab (HT), carriage
+return (CR), and line feed (LF) characters between tokens.
+
+Commentary in block quotes is not normative.
+
+> Non-normative commentary.
+
+## Value
+
+A representation of any expressible value in an OCapN message.
+
+- _abstract-value_: _abstract-boolean_ / _abstract-integer_ /
+  _abstract-float64_ / _abstract-selector_ / _abstract-string_ /
+  _abstract-byte-array_ / _abstract-byte-array_ / _abstract-struct_ /
+  _abstract-list_ / _abstract-record_
+- _concrete-value_: _concrete-boolean_ / _concrete-integer_ /
+  _concrete-float64_ / _concrete-selector_ / _concrete-string_ /
+  _concrete-byte-array_ / _concrete-byte-array_ / _concrete-struct_ /
+  _concrete-list_ / _concrete-record_
+
+## Boolean
+
+A value that is either true or false.
+
+> Examples:
+> - `f` corresponds to `f`.
+> - `t` corresponds to `t`.
+
+- _abstract-boolean_: `f` / `t` :: Corresponding to false and true respectively.
+- _concrete-boolean_: `f` / `t`
+
+## Integer
+
+An arbitrary precision signed integer.
+
+> Examples:
+> - `42` corresponds to `42+`.
+> - `-1` corresponds to `1-`.
+> - `0` corresponds to `0+`.
+
+- _abstract-integer_: _sign_? _digit_+
+- _sign_: `+` / `-`
+- _concrete-integer_: _digit_+ _sign_ :: No preceding zeroes.
+
+## Float64
+
+An IEEE 754 64-bit floating point number.
+
+> Examples:
+> - `nan` corresponds to the bytes 44 (`"D"`), 7f, f8, 00, 00, 00, 00, 00, 00
+>   in hexadecimal.
+
+- _abstract-float64_: _abstract-float64-number_ / _sign_? `inf` / `nan`
+- _abstract-float64-number_: _sign_? _digit_* `.` _digit_* :: Corresponding to the nearest
+  expressible concrete IEEE 754 64-bit floating point number.
```

Comment:
This is a bit hand-wavy; how are ties resolved? It also allows a lone `.`, which seems wrong.
```suggestion
- _abstract-float64-number_: _abstract-integer_ `.` _digit_* /
  _sign_? `.` _digit_+ :: Corresponding to the nearest expressible concrete IEEE
  754 64-bit floating point number, rounding ties to even.
```